### ISSUE TITLE
Github App Auth with Github Enterprise 

### ISSUE THREAD

**Issue by jretterer-prol** [promhippie/github_exporter#228]
While attempting to authenticate to a Github enterprise instance while Github App authentication I am receiving the following error:

`level=error ts=2023-07-11T18:43:58.694703235Z collector=repo msg="Failed to fetch repos" name=jretterer/testrepo err="GET https://<redacted>/api/v3/repos/jretterer/testrepo: 401 Bad credentials []"`

I am using the container in a standalone capacity and I have set the following environment vars for the container (info shortened and redacted as necessary):

```
- GITHUB_EXPORTER_APP_ID=52
- GITHUB_EXPORTER_INSTALLATION_ID=365
- GITHUB_EXPORTER_PRIVATE_KEY=LS0tL...S0tLQo=
- GITHUB_EXPORTER_BASE_URL=https://<redacted>
```

Am I missing a setting?

**Comment by tboerger** [comment]
I just checked the code, the GitHub app integration is not available for GitHub enterprise so far, that would restructuring the code at https://github.com/promhippie/github_exporter/blob/9a3f95931f00c07f5a325403857fccc9e1cdb4a8/pkg/action/server.go#L49-L120 a little bit.

**Comment by jretterer-prol** [comment]
Looks like its getting further but have run into this error:

`level=error ts=2023-07-13T13:34:55.136827781Z collector=repo msg="Failed to fetch repos" name=jretterer/testrepo err="Get \"https://<redacted>/api/v3/repos/jretterer/testrepo\": could not refresh installation id 365's token: received non 2xx response status \"302 Found\" when fetching https://<redacted>/app/installations/365/access_tokens"`

Taking a look at the [auth documentation](https://docs.github.com/en/enterprise-server@3.9/apps/creating-github-apps/authenticating-with-a-github-app/authenticating-as-a-github-app-installation) for GitHub Apps in enterprise. It looks like the `access_tokens` call may need to include the `/api/v3/` string that is required in a lot of enterprise operations.

**Comment by tboerger** [comment]
Since I have never received a response from gitHub to get access to GitHub Enterprise for development of this I can not test this. Have you tried to set `GITHUB_EXPORTER_BASE_URL=https://github.example.com/api/v3`? That's at least also mentioned on https://github.com/bradleyfalzon/ghinstallation#github-enterprise-example which is the library we are using for GitHub applications.

**Comment by jretterer-prol** [comment]
Adding `/api/v3` to the base URL did the trick.  Might be good to add that to the documentation as it was no required when using just the token.

**Comment by tboerger** [comment]
I have published v2.3.0 which incluse app support for GHE.
